03week/ticTacToe.js

```diff
@@ -25,24 +25,74 @@ function printBoard() {
 
 function horizontalWin() {
   // Your code here
+  if(board[0][0]=== 'X'&& board[0][1]==='X' && board[0][2] === 'X'){
+    return true
+  } if(board[0][0]=== 'O'&& board[0][1]==='O' && board[0][2] === 'O'){
+    return true
+  } if(board[1][0]=== 'X'&& board[1][1]==='X' && board[1][2] === 'X'){
+    return true
+  } if(board[1][0]=== 'O'&& board[1][1]==='O' && board[1][2] === 'O'){
+    return true
+  } if(board[2][0]=== 'X'&& board[2][1]==='X' && board[2][2] === 'X'){
+    return true
+  } if(board[2][0]=== 'O'&& board[2][1]==='O' && board[2][2] === 'O'){
+    return true
+  }
 }
 
 function verticalWin() {
   // Your code here
+  if(board[0][0]=== 'X'&& board[1][0]==='X' && board[2][0] === 'X'){
+    return true
+  } if(board[0][0]=== 'O'&& board[1][0]==='O' && board[2][0] === 'O'){
+    return true
+  } if(board[0][1]=== 'X'&& board[1][1]==='X' && board[2][1] === 'X'){
+    return true
+  } if(board[0][1]=== 'O'&& board[1][1]==='O' && board[1][2] === 'O'){
+    return true
+  } if(board[2][0]=== 'X'&& board[2][1]==='X' && board[2][2] === 'X'){
+    return true
+  } if(board[2][0]=== 'O'&& board[2][1]==='O' && board[2][2] === 'O'){
+    return true
+  }
+
 }
 
 function diagonalWin() {
   // Your code here
+  if(board[0][0]=== 'X'&& board[1][1]==='X' && board[2][2] === 'X'){
+    return true
+  } if (board[0][0]=== 'O'&& board[1][1]==='O' && board[2][2] === 'O'){
+    return true
+  } if(board[0][2]=== 'X'&& board[1][1]==='X' && board[2][0] === 'X'){
+    return true
+  } if(board[0][2]=== 'O'&& board[1][1]==='O' && board[2][0] === 'O'){
+    return true
+  }
+
+
 }
 
 function checkForWin() {
   // Your code here
+  if(diagonalWin() || verticalWin() || horizontalWin()){
+    return true
+  }
+
 }
 
 function ticTacToe(row, column) {
   // Your code here
+  if(playerTurn==='X'){
```

Comment:
Need to check that the box is not already occupied. 
And that the value provided is validd